src/accelerate/commands/launch.py

```diff
@@ -1052,7 +1052,10 @@ def launch_command(args):
             warned.append("\t`--dynamo_backend` was set to a value of `'no'`")
             args.dynamo_backend = "no"
 
-    if args.num_cpu_threads_per_process is None:
+    if (
+        defaults is None
+        or (defaults is not None and defaults.compute_environment != ComputeEnvironment.AMAZON_SAGEMAKER)
```

Comment:
Can we put this in another variable before? The test is hard to read right now.

Reply:
Done.